Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -658,6 +658,81 @@ class InAppMessageViewTest: IntegrationTest {
         // If url is valid, check if `handleDeepLink` method is called
         XCTAssertFalse(deeplinkUtilMock.handleDeepLinkCalled)
     }
+
+    // MARK: - Send events to Gist event listeners
+
+    // errorWithMessage
+    @MainActor
+    func test_givenAttemptToShowInlineMessage_EventListenerSet_expectFailToShow_expectRouteError() async {
+        messagingInAppImplementation.setEventListener(eventListenerMock)
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
+
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
+        await onDoneRenderingInAppMessageWithError(givenInlineMessage, insideOfInlineView: inlineView)
+
+        // errorWithMessage called
+        XCTAssertTrue(eventListenerMock.errorWithMessageCalled)
+        XCTAssertEqual(eventListenerMock.errorWithMessageCallsCount, 1)
+        XCTAssertEqual(eventListenerMock.errorWithMessageReceivedArguments, InAppMessage(gistMessage: givenInlineMessage))
+
+        // Other listeners not called
+        XCTAssertFalse(eventListenerMock.messageActionTakenCalled)
+        XCTAssertFalse(eventListenerMock.messageShownCalled)
+        XCTAssertFalse(eventListenerMock.messageDismissedCalled)
+    }
+
+    @MainActor
+    func test_givenAttemptToShowInlineMessageFails_expectMessageNotShown() async {
```

Comment:
Good test case. 

For organization purposes, I think this function belongs with [these tests](https://github.com/customerio/customerio-ios/blob/7a77b18fa660ed3383280812deeaf918e9d80813/Tests/MessagingInApp/Views/InAppMessageViewTest.swift#L76) in the file.

Reply:
The test case is in the same file as you linked above. 🤔

Reply:
Yes, it is the same file. 

This suggestion is that the test `test_givenAttemptToShowInlineMessageFails_expectMessageNotShown` is more related to the behavior of displaying in-app messages when an error happens. Not related to event listeners. So for organization purposes, I was suggesting to move the file to a different `// MARK:` section in the file such as `// MARK: Display in-app message`